### PR TITLE
servers: Add nicer error messages for plugin load fail

### PR DESCRIPTION
**Pull request by brianlheim**
- Add a new header (ErrorMessage.h) in include/server/ containing error
formatting messages for mismatched API and missing API version numbers.
- The mismatched API function gives helpful info on how to resolve the
issue, including download links and disambiguation text.
- Use these messages in supernova and scsynth.
- This will hopefully reduce headache of a thousand people posting "what does API mismatch mean" on the mailing list.

Example output:
```
ERROR: API version mismatch: /Users/brianheim/Library/Application Support/SuperCollider/Extensions/SC3plugins/SkUGens/FM7.scx
    This plugin was compiled for an older version of the plugin interface (2) than what you are currently using (3).
    Please use an older version of SuperCollider, which can be obtained from http://supercollider.github.io/, or use a newer version of this plugin.
    Note that plugin interface version numbers are separate from SuperCollider versioning.
    For example, an plugin interface version of 2 does not mean you need SuperCollider 2.
    Releases of sc3-plugins can be downloaded from https://github.com/supercollider/sc3-plugins/releases
    API version 3 was first required in SuperCollider 3.9.0.
```

Note that the "releases of sc3-plugins..." line only appears if you're compiling a path that contains the string "SC3plugins", which is the name of the sc3-plugins folder when you unzip one of the releases. Similarly, the line about API version 3 only appears if you're using v3 plugins or post-3.9.0 server.

### CODE REVIEW

**Comment by joshpar** [comment]
looks great to me.

**Comment by brianlheim** [comment]
I reworded the second sentence. The full message now reads:
```
ERROR: API version mismatch: /Users/brianheim/Library/Application Support/SuperCollider/Extensions/SC3plugins/SkUGens/FM7.scx
    This plugin was compiled for an older version of the plugin interface (2) than what you are currently using (3).
    Please use a newer version of this plugin, or an older version of SuperCollider, which can be downloaded from http://supercollider.github.io/
    Note that plugin interface version numbers are separate from SuperCollider versioning.
    For example, an plugin interface version of 2 does not mean you need SuperCollider 2.
    Releases of sc3-plugins can be downloaded from https://github.com/supercollider/sc3-plugins/releases
    API version 3 was first required in SuperCollider 3.9.0.
```

**Comment by brianlheim** [comment]
After some discussion, it got revised to this:

```
ERROR: API version mismatch: /Users/brianheim/Library/Application Support/SuperCollider/Extensions/SC3plugins/SkUGens/FM7.scx
    This plugin is not compatible with SuperCollider >=3.9.0
    The plugin has not been loaded; please find or compile a newer version.
    Releases of sc3-plugins can be downloaded from https://github.com/supercollider/sc3-plugins/releases
    (Plugin's API version: 2. Expected: 3)
```

**Comment by redFrik** [comment]
good initiative.  two questions... will this message also be printed for non sc3-plugins plugins?  then i don't agree with the github link in there.  and will the message be printed ~100 times or just once?  if multiple times then that's a lot of text.  at the moment i get ~100 single lines like *** ERROR: API Version Mismatch: /asdf/Library/Application Support/SuperCollider/Extensions/SC3plugins/ChaosUGens/PredPrey.scx

**Comment by brianlheim** [comment]
@redFrik 
> will this message also be printed for non sc3-plugins plugins?

From the OP:

"Note that the "releases of sc3-plugins..." line only appears if you're compiling a path that contains the string "SC3plugins", which is the name of the sc3-plugins folder when you unzip one of the releases. "

> and will the message be printed ~100 times or just once?

It will be printed once per bad plugin; this error is not fatal error, and loading continues normally afterward. If there are several plugins in several separate packages that are all outdated, users should know what they all are rather than just the first bad one.

**Comment by redFrik** [comment]
thanks @brianlheim.  sorry i missed that initial note about the github link. that's an awesome feature.  for the repetitions i was mainly thinking about the scide postwindow history and if 5 lines per error would over-flood. turns out 'scrollback' defaults to 1000 so no (they'll use up ~500 lines). and most likely not an issue anyway as people will update the first time they see these errors. i approve.

File: include/server/ErrorMessage.h [hunk 1]
```diff
@@ -0,0 +1,120 @@
+/**
+ *  \file ErrorMessage.hpp
```
Comment (snappizz) [inline include/server/ErrorMessage.h]: mismatch between this line and filename

Comment (brianlheim) [inline include/server/ErrorMessage.h]: fixed - renamed the file

**Comment by brianlheim** [comment]
@redFrik thanks! the critical eye is very much appreciated. :)

File: include/server/ErrorMessage.h (hunk 1 above)
Comment (snappizz) [inline include/server/ErrorMessage.h]: now it doesn't build because the other files still `#include "ErrorMessage.h"` :stuck_out_tongue_winking_eye:

Comment (brianlheim) [inline include/server/ErrorMessage.h]: :( ok, amended

**Pull request merged**